provider/scaleway/scaleway_discover.go

```diff
@@ -61,16 +61,18 @@ func (p *Provider) Addrs(args map[string]string, l *log.Logger) ([]string, error
 }
 
 func filterServersForTagName(servers *[]api.ScalewayServer, tagName string, l *log.Logger) []string {
-	var serverAddrs []string
-	for _, server := range *servers {
-		if stringInSlice(tagName, server.Tags) {
-			l.Printf("[INFO] discover-scaleway: Found server (%s) - %s with private IP: %s",
-				server.Name, server.Hostname, server.PrivateIP)
-			serverAddrs = append(serverAddrs, server.PrivateIP)
+	var addrs []string
+	if servers != nil {
+		for _, server := range *servers {
+			if stringInSlice(tagName, server.Tags) {
+				l.Printf("[DEBUG] discover-scaleway: Found server (%s) - %s with private IP: %s",
+					server.Name, server.Hostname, server.PrivateIP)
+				addrs = append(addrs, server.PrivateIP)
+			}
 		}
 	}
-
-	return serverAddrs
+	l.Printf("[DEBUG] discover-scaleway: Found ip addresses: %v", addrs)
```

Comment:
Please move this comment into the `Discover` function just before you return results.